tests/infer/mcmc/test_hmc.py

```diff
@@ -234,3 +234,69 @@ def model(data):
         posterior.append(trace.nodes['p_latent']['value'])
     posterior_mean = torch.mean(torch.stack(posterior), 0)
     assert_equal(posterior_mean, true_probs, prec=0.02)
+
+
+def test_logistic_regression_with_dual_averaging():
+    dim = 3
+    true_coefs = Variable(torch.arange(1, dim+1))
+    data = Variable(torch.randn(2000, dim))
+    labels = dist.Bernoulli(logits=(true_coefs * data).sum(-1)).sample()
+
+    def model(data):
+        coefs_mean = Variable(torch.zeros(dim), requires_grad=True)
+        coefs = pyro.sample('beta', dist.Normal(coefs_mean, Variable(torch.ones(dim))))
+        y = pyro.sample('y', dist.Bernoulli(logits=(coefs * data).sum(-1)), obs=labels)
+        return y
+
+    hmc_kernel = HMC(model, step_size=None)
+    hmc_kernel._adapted = True
+    hmc_kernel._trajectory_length = 1
```

Comment:
Why do we need to change this from its default value of `2\pi`? If we have to specify this for certain problems, we should pass it in the constructor directly, and document this argument. Also, we will need to think carefully about how some of these arguments will clash - e.g. we should not specify both the `trajectory_length` and `num_steps`. Any ideas how Stan models are able to run by just using the default value?